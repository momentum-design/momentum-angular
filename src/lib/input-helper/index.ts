export * from './input-helper.module';
export * from './input-helper.component';
