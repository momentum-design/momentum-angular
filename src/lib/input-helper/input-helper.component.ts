import { Component, OnInit, Input } from '@angular/core';

@Component({
  selector: 'md-input-helper',
  template: `
    <p [ngClass]="inputHelperClasses" class="md-input__help-text">
      {{ message }}
    </p>
  `,
  styles: [],
})
export class InputHelperComponent implements OnInit {
  @Input() public message: string;
  @Input() public className: string;

  constructor() {}

  ngOnInit() {}

  get inputHelperClasses() {
    return {
      [this.className]: this.className,
    };
  }
}
