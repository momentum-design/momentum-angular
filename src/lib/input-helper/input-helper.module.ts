import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { InputHelperComponent } from './input-helper.component';

@NgModule({
  declarations: [InputHelperComponent],
  imports: [CommonModule],
  exports: [InputHelperComponent],
})
export class InputHelperModule {}
