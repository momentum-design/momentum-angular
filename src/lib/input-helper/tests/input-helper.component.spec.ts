import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { InputHelperComponent } from '../input-helper.component';

describe('InputHelperComponent', () => {
  let component: InputHelperComponent;
  let fixture: ComponentFixture<InputHelperComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [InputHelperComponent],
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(InputHelperComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create and match snapshot', () => {
    expect(component).toBeTruthy();
    expect(fixture).toMatchSnapshot();
  });

  it('should render a help text to the input', () => {
    component.message = 'Help Text Message';
    fixture.detectChanges();

    const inputNativeElement = fixture.nativeElement;
    const helpText = inputNativeElement.querySelector('.md-input__help-text');

    expect(fixture).toMatchSnapshot();
    expect(helpText.textContent).toBe(' Help Text Message ');
  });
});
