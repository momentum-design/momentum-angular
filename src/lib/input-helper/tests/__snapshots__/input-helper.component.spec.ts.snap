// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputHelperComponent should create and match snapshot 1`] = `
<md-input-helper>
  <p
    class="md-input__help-text"
    ng-reflect-klass="md-input__help-text"
    ng-reflect-ng-class="[object Object]"
  >

  </p>
</md-input-helper>
`;

exports[`InputHelperComponent should render a help text to the input 1`] = `
<md-input-helper
  message={[Function String]}
>
  <p
    class="md-input__help-text"
    ng-reflect-klass="md-input__help-text"
    ng-reflect-ng-class="[object Object]"
  >
     Help Text Message
  </p>
</md-input-helper>
`;
